tests/non_functional_test.go

```diff
@@ -0,0 +1,89 @@
+package tests
+
+import (
+	"fmt"
+	"log"
+	"strconv"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+const (
+	splunkWaitTime  = time.Second * 60
+	minIngestRateMb = 1
+	minIngestLag    = 1
+)
+
+type Statistic struct {
+	EarliestTime string `json:"earliest_time"`
+	LatestTime   string `json:"latest_time"`
+	TotalEvents  string `json:"total_events"`
+}
+
+func TestNonFunctional(t *testing.T) {
+	topicName := "kafka-perf-test"
+	index := "kafka"
+	sourcetype := "otel-perf-tests"
+	source := "otel"
+	configFileTemplate := "basic_test.yaml.tmpl"
+	numOfMsg := 1_000_000
+	msgSize := 100
+
+	replacements := map[string]any{
+		"KafkaBrokerAddress": GetConfigVariable("KAFKA_BROKER_ADDRESS"),
+		"KafkaTopicName":     topicName,
+		"SplunkHECToken":     GetConfigVariable("HEC_TOKEN"),
+		"SplunkHECEndpoint":  fmt.Sprintf("https://%s:8088/services/collector", GetConfigVariable("HOST")),
+		"Source":             source,
+		"Index":              index,
+		"Sourcetype":         sourcetype,
+	}
+
+	configFileName := prepareConfigFile(t, configFileTemplate, replacements, configFilesDir)
+	connectorHandler := startOTelKafkaConnector(t, configFileName, configFilesDir)
+	defer stopOTelKafkaConnector(t, connectorHandler)
+
+	addKafkaTopic(t, topicName, 1, 1)
+	firstMsgSendTime, lstMsgSendTime := sendRandomizedMessages(topicName, numOfMsg, msgSize)
+	fmt.Printf("It took kafka client %f seconds to send all events\n", lstMsgSendTime.Sub(firstMsgSendTime).Seconds())
+
+	fmt.Printf("Waiting %f seconds for splunk to ingest all data\n", splunkWaitTime.Seconds())
+	time.Sleep(splunkWaitTime)
```

Comment:
is there a reason why do we wait here for a fixed time? wouldn't `require.Eventually` later on in a code be enough?